client/src/components/Common/Map/index.js

```diff
@@ -0,0 +1,95 @@
+import React from "react";
+import styled from "styled-components";
+import GoogleMapReact from 'google-map-react'
+import { HiLocationMarker } from 'react-icons/hi';
+
+import { borderRadius, red1 } from '../../../resources/style-constants'
+
+const Container = styled.div`
+  width: 100%;
+  height: 30vw;
+  border-radius: ${borderRadius};
+  overflow: hidden;
+`;
+
+const MapContainer = styled.div`
+  height: 100%;
+  width: 100%;
+  border-radius: ${borderRadius};
+`;
+
+const MapPinContainer = styled.div`
+  width: 128px;
+  color: ${red1};
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+`;
+
+const MapPinIcon = styled.div`
+  height: 48px;
+  width: 48px;
+`;
+
+const MapPinLabel = styled.h4`
+  margin: 0;
+  padding: 0;
+  line-height: 0;
+`;
+
+const InvalidMapContainer = styled.div`
+  height: 100%;
+  width: 100%;
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+  align-items: center;
+`;
+
+const MapPin = ({ text }) => {
+  return (
+    <MapPinContainer>
+      <MapPinIcon as={HiLocationMarker} />
+      <br />
+      <MapPinLabel>{text}</MapPinLabel>
+    </MapPinContainer>
+  );
+}
+
+/**
+ * @coords 
+ * ({ lat: 0, lng: 0 }) The coordinates of the the destination.
+ * 
+ * @zoomLevel
+ * (number) How far the map will be zoomed in on start.
+ * 
+ * @locationLabel
+ * (string) The text that appears below the map pin.
+ */
+const Map = ({ location, zoomLevel, locationLabel }) => {
```

Comment:
If you move your definitions of DEFAULT_LOCATION and DEFAULT_ZOOM above this line, inside of props you can write ({ location = DEFAULT_LOCATION, zoomLevel = DEFAULT_ZOOM, locationLabel}). So if location or zoomLevel isn't provided, it will use the default. 

This will also simplify your code down below because you won't have to use the nullish coalescing operator.

Reply:
Oooo neat! I'll do that